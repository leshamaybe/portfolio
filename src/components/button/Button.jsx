import React from 'react';
import Link from 'next/link';
import cn from 'classnames';

const Button = ({ link = '/', classname }) => {
    return (
        <Link href={link} className={cn('py-4 px-5 text-[14px] tracking-[2px] uppercase cursor-pointer', classname)}>
            VIEW PROJECT
        </Link>
    );
};

export default Button;
