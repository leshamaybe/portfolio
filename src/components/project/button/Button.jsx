import React from 'react';
import Link from 'next/link';
import s from './button.module.scss';
import cn from 'classnames';

const Button = ({ link, classname }) => {
    return (
        <Link href={link} className={cn(s.btn, classname)}>
            VIEW PROJECT
        </Link>
    );
};

export default Button;
