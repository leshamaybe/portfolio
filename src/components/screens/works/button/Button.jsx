import React from 'react';
import Link from 'next/link';
import s from './button.module.scss';
import cn from 'classnames';

const Button = ({ link }) => {
    return (
        <Link href={link} className={cn(s.btn, s['btn-1'])}>
            VIEW PROJECT
        </Link>
    );
};

export default Button;
