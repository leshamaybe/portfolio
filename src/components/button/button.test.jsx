import Button from './Button';
import '@testing-library/jest-dom';
import { render } from '@testing-library/react';

describe('Button', () => {
    it('Button renders correctly', async () => {
        const { container } = render(<Button />);
        expect(container).toMatchSnapshot();
    });
});
