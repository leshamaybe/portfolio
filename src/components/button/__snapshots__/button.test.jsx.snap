// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Button renders correctly 1`] = `
<div>
  <a
    class="py-4 px-5 text-[14px] tracking-[2px] uppercase cursor-pointer"
    href="/"
  >
    VIEW PROJECT
  </a>
</div>
`;
